services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/client/pulsar/DataPlaneClientImpl.java

```diff
@@ -18,24 +18,151 @@
+
+        return multicastTopics;
+    }
+
+    private MulticastGoalStateByte buildMulticastGoalStateByte(MulticastGoalState multicastGoalState) {
+        MulticastGoalStateByte multicastGoalStateByte = new MulticastGoalStateByte();
+        multicastGoalStateByte.setNextTopics(multicastGoalState.getNextTopics());
+        multicastGoalStateByte.setGoalStateByte(multicastGoalState.getGoalState().toByteArray());
+
+        return multicastGoalStateByte;
+    }
+
+    private void createGoalState(MulticastGoalState multicastGoalState) throws Exception {
+        Map<String, List<String>> multicastTopics = getMulticastTopics(multicastGoalState.getHostIps());
+
+        for (Map.Entry<String, List<String>> entry: multicastTopics.entrySet()) {
+            String multicastTopic = entry.getKey();
+            List<String> groupTopics = entry.getValue();
+
+            multicastGoalState.setNextTopics(groupTopics);
+
+            Producer<MulticastGoalStateByte> producer = pulsarClient
+                    .newProducer(JSONSchema.of(MulticastGoalStateByte.class))
+                    .topic(multicastTopic)
+                    .enableBatching(false)
+                    .create();
+
+            producer.send(buildMulticastGoalStateByte(multicastGoalState));
+
+            LOG.info("Send multicastGoalState to topic:{} success, " +
+                            "groupTopics: {}, unicastGoalStates: {}",
+                    multicastTopic, groupTopics, multicastGoalState);
+        }
+    }
+
     @Override
     public List<Map<String, List<Goalstateprovisioner.GoalStateOperationReply.GoalStateOperationStatus>>> createGoalStates(List<UnicastGoalState> unicastGoalStates) throws Exception {
-        return null;
+        for (UnicastGoalState unicastGoalState: unicastGoalStates) {
+            String nextTopic = topicManager.getGroupTopicByHostIp(unicastGoalState.getHostIp());
+            if (StringUtils.isEmpty(nextTopic)) {
+                LOG.error("Can not find next topic by host ip:{}", unicastGoalState.getHostIp());
+                throw new GroupTopicNotFound();
+            }
+
+            String topic = nextTopic;
+            String unicastTopic = topicManager.getUnicastTopic();
+            if (!StringUtils.isEmpty(unicastTopic)) {
+                unicastGoalState.setNextTopic(nextTopic);
+                topic = unicastTopic;
+            }
+
+            Producer<UnicastGoalStateByte> producer = pulsarClient
+                    .newProducer(JSONSchema.of(UnicastGoalStateByte.class))
+                    .topic(topic)
+                    .enableBatching(false)
+                    .create();
+            producer.send(buildUnicastGoalStateByte(unicastGoalState));
+
+            LOG.info("Send unicastGoalStates to topic:{} success, " +
+                    "unicastGoalStates: {}", nextTopic, unicastGoalState);
+        }
+
+        List<Goalstateprovisioner.GoalStateOperationReply.GoalStateOperationStatus> tempList = new ArrayList<>();
```

Comment:
I know we don't have a response from MQ for now. @chenpiaoping Let us plan to design a feedback channel here: it would be ideal if we can reuse the same topic with different key.